src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -378,22 +372,25 @@ internal static void AdjustPosition()
 		bool cursorTooLow = false;
 
 		// Find the next parent ScrollView that is scrollable
-		var superView = View.FindResponder<UIScrollView>();
+		var superView = View.FindResponder<UIScrollView>() ?? View as UIScrollView;
 		var superScrollView = FindParentScroll(superView);
 
 		CGRect? superScrollViewRect = null;
 		var topBoundary = topLayoutGuide;
 		var bottomBoundary = (double)keyboardYPosition;
 
-		if (superScrollView is not null){
+		if (superScrollView is not null)
+		{
 			superScrollViewRect = superScrollView.ConvertRectToView(superScrollView.Bounds, window);
-			topBoundary = Math.Max(topBoundary, superScrollViewRect.Value.Top + TextViewDistanceFromTop);
-			bottomBoundary = Math.Min(bottomBoundary, superScrollViewRect.Value.Bottom - TextViewDistanceFromBottom);
+			topBoundary = Math.Max(topBoundary, superScrollViewRect.Value.Top);
+			bottomBoundary = Math.Min(bottomBoundary, superScrollViewRect.Value.Bottom - bottomBuffer);
 		}
 
+		bool forceSetContentInsets = false;
+
 		// scenario where we go into an editor with the "Next" keyboard button,
 		// but the cursor location on the editor is scrolled below the visible section
-		if (View is UITextView && cursorRect.Y >= viewRectInWindow.GetMaxY())
+		if (View is UITextView && IsKeyboardShowing && cursorRect.Y >= viewRectInWindow.GetMaxY())
```

Comment:
Please use `View is UITextView or UITextField` instead in order to include 3rd party input controls